exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/metadata/Metadata.java

```diff
@@ -353,22 +353,15 @@ private static boolean ignoreReadingMetadata(MetadataContext metaContext, Path p
     writeFile(metadataTableWithRelativePaths.getSummary(), new Path(path, METADATA_SUMMARY_FILENAME), fs);
     Metadata_V4.MetadataSummary metadataSummaryWithRelativePaths = metadataTableWithRelativePaths.getSummary();
 
-    if (directoryList.size() > 0 && childFiles.size() == 0) {
-      ParquetTableMetadataDirs parquetTableMetadataDirsRelativePaths =
-          new ParquetTableMetadataDirs(metadataSummaryWithRelativePaths.directories);
-      writeFile(parquetTableMetadataDirsRelativePaths, new Path(path, METADATA_DIRECTORIES_FILENAME), fs);
-      if (timer != null) {
-        logger.debug("Creating metadata files recursively took {} ms", timer.elapsed(TimeUnit.MILLISECONDS));
-      }
-      ParquetTableMetadataDirs parquetTableMetadataDirs = new ParquetTableMetadataDirs(directoryList);
-      return Pair.of(parquetTableMetadata, parquetTableMetadataDirs);
-    }
-    List<Path> emptyDirList = new ArrayList<>();
+    ParquetTableMetadataDirs parquetTableMetadataDirsRelativePaths =
```

Comment:
Pls add a comment indicating that and empty directories list will be created for leaf level directories.  For sub-directories with mix of files and directories, the directories will be tracked.

Reply:
Added a comment.